### PR TITLE
crosstool-ng HEAD version

### PR DESCRIPTION
**Pull request by chrisballinger**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Adds option for crosstool-ng to be build with HEAD option.

### CODE REVIEW

File: Formula/crosstool-ng.rb [hunk 1]
```diff
@@ -12,6 +13,7 @@ class CrosstoolNg < Formula
     sha256 "c119b94b8b4782935e7c2968d195cddcfd2faa089768c0a0bf84dfcb8d5713cc" => :yosemite
   end
 
+  depends_on "bash" => :build
```
Comment (fxcoudert) [inline Formula/crosstool-ng.rb]: We definitely do not want that. Moreover, it ends up hardcoding the path to Homebrew bash in the installed version.

**Comment by fxcoudert** [comment]
The [bootstrap script](https://github.com/crosstool-ng/crosstool-ng/blob/master/bootstrap) requires bash 4. We *strongly* prefer formulas to build with the system shell. As long as it's head-only, I'd recommend we don't include HEAD in the formula.

**Comment by ilovezfs** [comment]
Agreed with @fxcoudert. Thanks for the PR nonetheless @chrisballinger

**Comment by chrisballinger** [comment]
Ah ok, no problem. There's another problem anyway that should require a "caveat" section, where it won't run unless you say `make -f ct-ng` because macOS interprets it as a shell file and not a makefile. This applies to the release version as well.

Additionally, I wasn't able to get it to successfully build a toolchain anyway, but that's a problem for upstream.

**Pull request closed without merging**